config/routes.rb

```diff
@@ -1,6 +1,10 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
+  get "/users", to: "users#index", as: "users"
+  get "/users/:id", to: "users#show", as: 'user'
+  get "/users/:user_id/posts", to: "posts#index", as: 'user_posts'
+  get "/users/:user_id/posts/:id", to: "posts#show", as: 'user_post'
```

Comment:
- [ **OPTIONAL** ] Good job implementing all the requested routes. However, what do you think if if you use the `resources` method to define your routes as it is the easy one more understandable by all users. For this will only need to add the `action` for each route you want.
Here is the example of all the above routes:
 ```ruby
 resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

Reply:
thank you @tresorsawasawa , i will make changes.